src/retrieval/reranking_algorithms.py

```
@@ -0,0 +1,243 @@
+                        normalized_score * 0.7  # Cross-encoder score
+                    )
+                
+                logger.debug(f"Cross-encoder reranking completed for {len(results)} results")
+            else:
+                # Fallback to enhanced heuristic scoring
+                self._enhanced_heuristic_rerank(results, query)
+                
+        except ImportError:
+            logger.warning("sentence-transformers not available, using enhanced heuristic reranking")
+            self._enhanced_heuristic_rerank(results, query)
+        except Exception as e:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)